plugins/modules/grafana_organization.py

```diff
@@ -0,0 +1,204 @@
+        if status_code == 404:
+            return None
+        elif status_code == 401:
+            self._module.fail_json(failed=True, msg="Unauthorized to perform action '%s' on '%s' header: %s" % (method, full_url, self.headers))
+        elif status_code == 403:
+            self._module.fail_json(failed=True, msg="Permission Denied")
+        elif status_code == 200:
+            return self._module.from_json(resp.read())
+        
+        if resp is None:
+            self._module.fail_json(failed=True, msg="Cannot connect to API Grafana %s" % info['msg'] , status=status_code, url=info['url'])
+        else:
+            self._module.fail_json(failed=True, msg="Grafana Org API answered with HTTP %d" % status_code, body=self._module.from_json(resp.read()))
+
+    def get_actual_org(self, name):
+        # https://grafana.com/docs/grafana/latest/http_api/org/#get-organization-by-name
+        url = "/api/orgs/name/{name}".format(name=name)
+        return self._send_request(url, headers=self.headers, method="GET")
+
+    def create_org(self, name):
+        # https://grafana.com/docs/http_api/org/#create-organization
+        url = "/api/orgs"
+        org = dict(name=name)
+        self._send_request(url, data=org, headers=self.headers, method="POST")
+        return self.get_actual_org(name)
+
+    def delete_org(self, org_id):
+        # https://grafana.com/docs/http_api/org/#delete-organization
+        url = "/api/orgs/{org_id}".format(org_id=org_id)
+        return self._send_request(url, headers=self.headers, method="DELETE")
+
+
+def setup_module_object():
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_together=base.grafana_required_together()
+    )
+    return module
+
+
+argument_spec = base.grafana_argument_spec()
+argument_spec.update(
+    state=dict(choices=['present', 'absent'], default='present'),
+    name=dict(type='str', required=True),
+)
+
+
+def main():
+    module = setup_module_object()
+    state = module.params['state']
+    name = module.params['name']
+
+    grafana_iface = GrafanaUserInterface(module)
+
+    # search org by name
+    actual_org = grafana_iface.get_actual_org(name)
+    if state == 'present':
+        has_changed = False
+
+        if actual_org is None:
+            # create new org
+            actual_org = grafana_iface.create_org(name)
+            has_changed = True
+            module.exit_json(changed=has_changed,msg='Organization %s created.' % name, org=actual_org)
+        else:
+            module.exit_json(changed=has_changed,msg='Organization %s already created.' % name, org=actual_org)
+
+    elif state == 'absent':
+        if actual_org is None:
+            module.exit_json(msg="No org found, nothing to do")
+        # delete org
+        result = grafana_iface.delete_org(actual_org.get("id"))
+        module.exit_json(changed=True, msg=result.get("message"), org=actual_org)
```

Comment:
Any reason to provide the org  parameter here while it has been removed ?

Reply:
You're wright, i delete this part.